pytorch/pytorch
# Implement initial version of autograd with named tensors

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#25604 Implement initial version of autograd with named tensors**

In this initial version:
- autograd ignores all names.
- tensor.grad is unnamed, unless the user manually assigns to it.
- if a grad tensor has any names, perhaps the user was hoping for some
alignment-checking behavior that named tensor offers for other ops. We
raise a warning in this case.

Future: do some more extensive checking to see if this actually works in
all cases.

Test Plan:
- [namedtensor ci]
- Check a warning is raised if a grad tensor has names.
- Check tensor.grad field is unnamed.
- Check that we can perform backward on an op that doesn't explictly
support names in backward. `sigmoid` is one such op.

Differential Revision: [D17171788](https://our.internmc.facebook.com/intern/diff/D17171788)
